### DISCUSSION TITLE
When updating running cluster (adding a master for instance) playbook fails

### DISCUSSION THREAD

**Discussion by gcraenen** [techno-tim/k3s-ansible#185] [Questions]
When I try to update a running cluster, for instance when I want to add an new master node, I change the hosts.ini file and run the playbook. It then fails at the step "TASK [k3s/master : Verify that all nodes actually joined (check k3s-init.service if this fails)] ***".

Keeps running through the 20 tries and keeps failing.

I don't want to reset and clean out the whole running cluster for an update, because there are several deployments running in it.

**Comment by gcraenen** [comment]
Is there another way to add a master or worker node to the already running cluster, that was started with this ansible playbook?

**Comment by timothystewart6** [comment]
Did you add the new node to your `hosts.ini` or remove them all but the new one?

**Reply by gcraenen** [reply]
I added it to the orignal list of masters. It is an Odroid I want to add as master and then after that remove another master that is not functioning ok (constant diskpressure, etc. signals in Rancher). Removing a node with kubectl or in Rancher is no problem at all, but adding nodes is giving me headaches.

I don't want to tear down the cluster and start it again, because I have several apps and deployments running in it. Like the complete "arr" mediasuite with emby, pihole, etc. I can share those for your library if you like.

**Reply by timothystewart6** [reply]
I would recommend just adding and removing the manual way for now, without the playbook

**Reply by gcraenen** [reply]
OK, thanks! I'll find out how that's done. The removing part is easy via kubectl or rancher. The adding part is more difficult.

**Reply by fabio-anzola** [reply]
@gcraenen did you manage to add a new master/worker node? If so, could you share your steps. I'm in a similar situation right now.

Thanks in advance!
